adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/amqp/AmqpPublisher.java

```diff
@@ -39,21 +39,7 @@ public void publishImportFailure(Long id, String errMsg) {
   }
 
   private void publish(String topic, Object message) {
-    for (int retries = properties.getAmqp().getPublishRetries(); retries >= 0; retries--) {
-      try {
-        this.rabbitTemplate.convertAndSend(properties.getAmqp().getExchange(), topic, message);
-        return;
-      } catch (AmqpException e) {
-        try {
-          Thread.sleep(properties.getAmqp().getPublishBackoff());
-        } catch (InterruptedException interruptedException) {
-          Thread.currentThread().interrupt();
-          throw new RuntimeException(interruptedException);
-        }
-        log.warn("Message publish failed ({}). Retrying in {}", retries, properties.getAmqp().getPublishBackoff());
-      }
-    }
-    log.error("Sending message {} to topic {} failed", message.toString(), topic);
+    var event = new OutboxEvent(topic, message);
```

Comment:
I assume that the calling class will manage the transaction then? Just curious...

Reply:
No, transactions are handled by the `@Transactional` annotations in the `DatasourceManager`. For example here: https://github.com/jvalue/open-data-service/blob/1e670cdf2d36aae12273a2a965e0dd3e84edfe21/adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java#L49-L56